interface1.js

```diff
@@ -884,35 +884,154 @@ window.addEventListener('load', function(){
 			genStringsList = fixedStringList;
 		}
 
-		var length = spotForm.genStringsInput.length;
-		if(length === undefined) {
-			length = 1;
+		/* minProblem, maxProblem, and inputPresent are flags for specific tests. The default value is 0. If the value equals to 1, then the test failed */
+		var inputFive = false;
+		var minOrMaxProblem = false;
+		var inputPresent = 0;
+		var problemError = "";
+
+		var numTerminalStrings = spotForm.genStringsInput.length;
+		if(numTerminalStrings === undefined) {
+			numTerminalStrings = 1;
+		}
+
+		/* checking if list of terminals input is empty */
+		for(var i=0; i<numTerminalStrings; i++){
+			inputPresent = (numTerminalStrings==1 ? spotForm.genStringsInput.value !== "": spotForm.genStringsInput[i].value !== "");
+			if (inputPresent == true){
+				inputPresent = 1;
+				break;
+			}
 		}
-		var inputList = spotForm.genStringsInput.value;
-		var min = spotForm.genStringsMin.value;
-		var max = spotForm.genStringsMax.value;
-
-		for(var i=0; i<length; i++){
-			if(length > 1) {
-				inputList = spotForm.genStringsInput[i].value;
-				min = spotForm.genStringsMin[i].value;
-				max = spotForm.genStringsMax[i].value;
+		/*if inputPresent is 0 then all the List of terminals are empty else if 1 then there
+		is at least one terminal input*/
+		if(inputPresent == 1){
+			breakloop:
+			for(var i=0; i<numTerminalStrings; i++){
+				if (numTerminalStrings > 1){
```

Comment:
add variables:
input, min, max that are set to either spotForm.genStringsInput (if numTerminalStrings ==1 ) or spotForm.genStringsInput[i] if numTerminalStrings > 1, and so on.
Then eliminate the if/else structure here with its code duplication.